### PR TITLE
WT-12701 Add ex_encrypt to examples-c-tsan

### PR DESCRIPTION
**Pull request by ajmorton**
Add `ex_encrypt` to `examples-c-tsan` and update WiredTiger to resolve warnings raised by running `ex_encrypt`.
This PR also contain additional changes that aren't required by `ex_encrypt`, but will be needed for the addition of future tests. I've tried to move some changes into this PR to balance the workload with future PRs.

### CODE REVIEW

File: src/include/cache_inline.h [hunk 1]
```diff
@@ -200,14 +206,16 @@ __wt_cache_bytes_image(WT_CACHE *cache)
 static WT_INLINE uint64_t
 __wt_cache_bytes_other(WT_CACHE *cache)
 {
-    uint64_t bytes_other;
+    uint64_t bytes_other, bytes_inmem, bytes_image_intl, bytes_image_leaf;
 
+    bytes_inmem = __wt_atomic_load64(&cache->bytes_inmem);
```
Comment (luke-pearson) [inline src/include/cache_inline.h:211]: I think we may need an interface for the cache which returns atomic loads and stores, or it locks every access effectively getters an setters. I think we can do it with function pointers, I'll think on it.

Comment (luke-pearson) [reply src/include/cache_inline.h:211]: Technically private interfaces are possible but they're not coming to WiredTiger without mega refactors so I'll close this comment.

File: src/include/cache_inline.h [hunk 2]
```diff
@@ -133,12 +133,14 @@ __wt_cache_pages_inuse(WT_CACHE *cache)
  *     Apply the cache overhead to a size in bytes.
  */
 static WT_INLINE uint64_t
-__wt_cache_bytes_plus_overhead(WT_CACHE *cache, uint64_t sz)
+__wt_cache_bytes_plus_overhead(WT_CACHE *cache, uint64_t *sz)
```
Comment (luke-pearson) [inline src/include/cache_inline.h:136]: I'd rather everything that goes in here is loaded atomically then passed in, burying the atomic here is very weird.

Comment (ajmorton) [reply src/include/cache_inline.h:136]: I had that originally, but wasn't happy with the verbosity of all the calls:
`__wt_cache_bytes_plus_overhead(cache, __wt_atomic_load64(&cache->bytes_internal))`
vs
__wt_cache_bytes_plus_overhead(cache, &cache->bytes_internal)

This approach saves us approx. 35 additional calls to atomic load. The tradeoff is some ugliness in functions like `__wt_btree_dirty_inuse` where we pass a reference to a stack variable in to be atomically read.

Comment (ajmorton) [reply src/include/cache_inline.h:136]: Example commit `atomic_load`ing before passing values into `__wt_cache_bytes_plus_overhead`: https://github.com/wiredtiger/wiredtiger/commit/94065fafc733ec6eb059679758c66fd59afe33b3
It's not as bad as I thought - just an additional 15 `atomic_load` calls. I have a soft preference for the current state of things but don't mind changing it if you want to push for it.

Comment (luke-pearson) [reply src/include/cache_inline.h:136]: I'll definitely push for it, the function shouldn't really go and atomic load inside of it. It should just add the overhead to the provided value.

Comment (ajmorton) [reply src/include/cache_inline.h]: 👍 I've added [94065fa](https://github.com/wiredtiger/wiredtiger/pull/10400/commits/94065fafc733ec6eb059679758c66fd59afe33b3) to this PR.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**